pkg/config/types/v1alpha1/config.go

```diff
@@ -24,33 +26,47 @@ const (
 )
 
 // Version implements the Configurator interface.
-func (n *Config) Version() string {
+func (c *Config) Version() string {
 	return Version
 }
 
 // Debug implements the Configurator interface.
-func (n *Config) Debug() bool {
+func (c *Config) Debug() bool {
 	return false
 }
 
 // Machine implements the Configurator interface.
-func (n *Config) Machine() machine.Machine {
-	return n.MachineConfig
+func (c *Config) Machine() machine.Machine {
+	return c.MachineConfig
 }
 
 // Cluster implements the Configurator interface.
-func (n *Config) Cluster() cluster.Cluster {
-	return n.ClusterConfig
+func (c *Config) Cluster() cluster.Cluster {
+	return c.ClusterConfig
 }
 
 // Validate implements the Configurator interface.
-func (n *Config) Validate() error {
+func (c *Config) Validate(mode runtime.Mode) error {
```

Comment:
This is a simple implementation for now. I plan to follow up with more robust checks and tests.